### PR TITLE
chore: Migrate CJS to ESM

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [x] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | chore: add devDependencies |
| 🇨🇳 Chinese | 迁移到 ts + esm 的一点准备工作，无需纳入 changelog |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 1cb7d5f</samp>

Improved TypeScript support for `antd-tools` and removed unused script `css-variable-sync.js`.

### <samp>🤖 Generated by Copilot at 1cb7d5f</samp>

*  Add type definition packages for `isomorphic-fetch`, `node`, `progress` and `puppeteer` modules to enable TypeScript support for `antd-tools` package ([link](https://github.com/ant-design/ant-design/pull/41809/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R178), [link](https://github.com/ant-design/ant-design/pull/41809/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L185-R188))
* Delete `scripts/css-variable-sync.js` file, which was a script for synchronizing CSS variables across components ([link](https://github.com/ant-design/ant-design/pull/41809/files?diff=unified&w=0#diff-0a94540713c69b716168ea85d46d1ea5f987db96a0a34d2cb359f995c6aa1659))

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -21,7 +21,8 @@
     "target": "es6",
     "lib": ["dom", "es2017"],
     "skipLibCheck": true,
-    "stripInternal": true
+    "stripInternal": true,
+    "resolveJsonModule": true
```
Comment (MadCcc) [inline tsconfig.json:25]: node 需要的加在 tsconfig.node.json 里

Comment (li-jia-nan) [reply tsconfig.json:25]: 我改一下

**Comment by MadCcc** [comment]
迁一个 script 看看效果，不然这些依赖加了感觉没意义

**Comment by li-jia-nan** [comment]
@SocketSecurity ignore @swc/core@1.3.51

**Comment by li-jia-nan** [comment]
@SocketSecurity ignore @swc/core@1.3.53

**Comment by li-jia-nan** [comment]
@SocketSecurity ignore @swc/core@1.3.55

**Review by MadCcc — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -55,11 +55,11 @@
     "clean-lockfiles": "rm -rf package-lock.json yarn.lock",
     "collect-token-statistic": "ts-node --project tsconfig.node.json scripts/collect-token-statistic.js",
     "token-meta": "node scripts/generate-token-meta.js",
-    "prestart": "npm run version & npm run collect-token-statistic & npm run token-meta",
-    "precompile": "npm run version & npm run collect-token-statistic & npm run token-meta",
+    "prestart": "npm run version && npm run collect-token-statistic && npm run token-meta",
```
Comment (zombieJ) [inline package.json:58]: 之前并行跑是因为会快点，不过的确偶尔会遇到文件还没生成的情况。这样改也好~

Comment (li-jia-nan) [reply package.json:58]: 现在并行跑每次都会遇到文件还没生成，100%复现的

**Pull request merged**